File: src/main/java/mediathek/gui/dialog/StandardCloseDialog.java
package mediathek.gui.dialog;

import mediathek.gui.actions.DisposeDialogAction;
import mediathek.tool.EscapeKeyHandler;

import javax.swing.*;
import java.awt.*;

/**
 * A standard swing dialog template with a close button.
 */
public abstract class StandardCloseDialog extends JDialog {
    public StandardCloseDialog(Frame owner, String title, boolean modal) {
        super(owner, title, modal);
        setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
        EscapeKeyHandler.installHandler(this, this::dispose);

        var contentPane = getContentPane();
        contentPane.setLayout(new BorderLayout());
        contentPane.add(createContentPanel(), BorderLayout.CENTER);
        var buttonPanel = new ButtonPanel();
        buttonPanel.add(createButtonPanel(),BorderLayout.EAST);
        contentPane.add(buttonPanel, BorderLayout.SOUTH);
        pack();
    }

    public abstract JComponent createContentPanel();

    public final ButtonFlowPanel createButtonPanel() {
        ButtonFlowPanel pnl = new ButtonFlowPanel();
        JButton btn = new JButton(new DisposeDialogAction(this, "Schließen", "Dialog schließen"));
        getRootPane().setDefaultButton(btn);
        pnl.add(btn);
        return pnl;
    }
}

File: src/main/kotlin/mediathek/gui/dialog/StandardCloseDialog.kt
package mediathek.gui.dialog

import mediathek.gui.actions.DisposeDialogAction
import mediathek.tool.EscapeKeyHandler
import java.awt.BorderLayout
import java.awt.Frame
import javax.swing.JButton
import javax.swing.JComponent
import javax.swing.JDialog

/**
 * A standard swing dialog template with a close button.
 */
abstract class StandardCloseDialog(owner: Frame?, title: String, modal: Boolean) : JDialog(owner, title, modal) {
    init {
        defaultCloseOperation = DISPOSE_ON_CLOSE
        EscapeKeyHandler.installHandler(this, ::dispose)

        contentPane.apply {
            layout = BorderLayout()
            add(createContentPanel(), BorderLayout.CENTER)
            add(
                ButtonPanel().apply { add(createButtonPanel(), BorderLayout.EAST) },
                BorderLayout.SOUTH,
            )
        }
        pack()
    }

    abstract fun createContentPanel(): JComponent

    private fun createButtonPanel(): ButtonFlowPanel {
        val button = JButton(DisposeDialogAction(this, "Schließen", "Dialog schließen"))
        rootPane.defaultButton = button
        return ButtonFlowPanel().apply { add(button) }
    }
}
